lib/membrane/rtp/demuxer.ex

```diff
@@ -0,0 +1,384 @@
+          |> Map.update!(
+            :pads_waiting_for_stream,
+            &Map.delete(&1, matching_pad_waiting_for_stream)
+          )
+
+        {[stream_format: {matching_pad_waiting_for_stream, %RTP{}}], state}
+    end
+  end
+
+  @spec find_matching_pad_for_stream(
+          ExRTP.Packet.t(),
+          %{Pad.ref() => stream_id()},
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: Pad.ref() | nil
+  defp find_matching_pad_for_stream(packet, pads_waiting_for_stream, payload_type_mapping) do
+    Enum.find(pads_waiting_for_stream, fn {_pad_ref, stream_id} ->
+      pad_stream_match?(stream_id, packet.ssrc, packet.payload_type, payload_type_mapping)
+    end)
+    |> case do
+      nil -> nil
+      {pad_ref, _stream_id} -> pad_ref
+    end
+  end
+
+  @spec find_matching_stream_for_pad(
+          stream_id(),
+          %{RTP.ssrc() => State.StreamState.t()},
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: RTP.ssrc() | nil
+  defp find_matching_stream_for_pad(stream_id, stream_states, payload_type_mapping) do
+    Enum.find(stream_states, fn {ssrc, stream_state} ->
+      pad_stream_match?(stream_id, ssrc, stream_state.payload_type, payload_type_mapping)
+    end)
+    |> case do
+      nil -> nil
+      {ssrc, _stream_state} -> ssrc
+    end
+  end
+
+  @spec pad_stream_match?(
+          stream_id(),
+          RTP.ssrc(),
+          RTP.payload_type(),
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: boolean()
+  defp pad_stream_match?(
+         pad_options_stream_id,
+         stream_ssrc,
+         stream_payload_type,
+         payload_type_mapping
+       ) do
+    case pad_options_stream_id do
+      {:ssrc, pad_ssrc} ->
+        stream_ssrc == pad_ssrc
+
+      {:payload_type, pad_payload_type} ->
+        stream_payload_type == pad_payload_type
+
+      {:encoding_name, pad_encoding_name} ->
+        %{payload_type: pad_payload_type} =
+          RTP.PayloadFormat.resolve(
+            encoding_name: pad_encoding_name,
+            payload_type_mapping: payload_type_mapping
+          )
+
+        stream_payload_type == pad_payload_type
+    end
+  end
+
+  @spec append_buffer_to_queued_buffers(RTP.ssrc(), Buffer.t(), State.t()) :: State.t()
+  defp append_buffer_to_queued_buffers(ssrc, buffer, state) do
+    Bunch.Struct.update_in(state, [:stream_states, ssrc, :queued_buffers], &[buffer | &1])
+  end
+
+  @spec put_packet_into_buffer(ExRTP.Packet.t(), RTP.PayloadFormat.payload_type_mapping()) ::
+          Membrane.Buffer.t()
+  defp put_packet_into_buffer(packet, payload_type_mapping) do
+    %{clock_rate: clock_rate} =
+      RTP.PayloadFormat.resolve(
+        payload_type: packet.payload_type,
+        payload_type_mapping: payload_type_mapping
+      )
+
+    pts =
+      if clock_rate == nil,
+        do: nil,
+        else: Ratio.new(packet.timestamp, clock_rate) |> Membrane.Time.seconds()
+
+    %Membrane.Buffer{
+      payload: packet.payload,
+      pts: pts,
```

Comment:
To know the PTS, we'd need to handle timestamp rollovers. That's in the jitter buffer. Here we can only set the RTP timestamp in metadata